### PR TITLE
Adding NNPBE

### PR DESCRIPTION
**Pull request by advikavs**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/dft/nnxc.py [hunk 1]
```diff
@@ -153,6 +153,68 @@ def get_edensityxc(
...
+            assert densinfo.u.grad is not None
+            assert densinfo.d.grad is not None
+            nu = densinfo.u.value.unsqueeze(-1)
+            nd = densinfo.d.value.unsqueeze(-1)
+            n = nu + nd  # (*BD, nr, 1)
+            n_offset = n + 1e-18  # avoiding nan
+            xi = (nu - nd) / n_offset
+            s = safenorm(densinfo.u.grad + densinfo.d.grad, dim=-1).unsqueeze(-1)
+
+        print("s_1", s.shape)
```
Comment (advikavs) [inline deepchem/models/dft/nnxc.py:197]: remove print after testing.

File: deepchem/models/dft/nnxc.py [hunk 2]
```diff
@@ -154,6 +155,62 @@ def get_edensityxc(
         return res
 
 
+class NNPBE(BaseNNXC):
+    # neural network xc functional of GGA (receives the density and grad as inputs)
```
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:159]: Can you add docs/comments? I will do a full review after that

File: deepchem/models/dft/nnxc.py [hunk 3]
```diff
@@ -154,6 +155,120 @@ def get_edensityxc(
...
+    Comprehensive Semiconductor Science and Technology,
+    Elsevier,
+    2011,
+    Pages 383-479,
+    ISBN 9780444531537,
+    https://doi.org/10.1016/B978-0-44-453153-7.00033-X.
+    """
+
+    def __init__(self, nnmodel: torch.nn.Module):
+        # nnmodel should receives input with shape (..., 3)
```
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:198]: Can you move this into the docstring instead of leaving as comments?

File: deepchem/models/dft/nnxc.py [hunk 4]
```diff
@@ -154,6 +155,120 @@ def get_edensityxc(
...
+        # where the last dimension is for:
+        # (0) total density (n): (n_up + n_dn), and
+        # (1) spin density (xi): (n_up - n_dn) / (n_up + n_dn)
+        # (2) normalized gradients (s): |del(n)| / [2(3*pi^2)^(1/3) * n^(4/3)]
+        # the output of the model must have shape of (..., 1)
+        # it represents the energy density per density per volume
+        super().__init__()
```
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:205]: This should come after the docstring

File: deepchem/models/dft/nnxc.py (hunk 3 above)
Comment (advikavs) [reply deepchem/models/dft/nnxc.py]: done.

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**